packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -720,14 +720,23 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
 
         periodicLogger.stop();
         // eslint-disable-next-line max-len
-        await this.setStatusMessage(`Finished! Total ${this.stats.state.requestsFinished + this.stats.state.requestsFailed} requests: ${this.stats.state.requestsFinished} succeeded, ${this.stats.state.requestsFailed} failed.`, { isStatusMessageTerminal: true });
+        await this.setStatusMessage(`Finished! Total ${stats.requestsTotal} requests: ${stats.requestsFinished} succeeded, ${stats.requestsFailed} failed.`, { isStatusMessageTerminal: true });
         return stats;
     }
 
     async getRequestQueue() {
-        this.requestQueue ??= await RequestQueue.open();
+        if (!this.requestQueue) {
```

Comment:
ok, now I get why you wanted to handle this elsewhere. even like this its not universal, the queue can be set explicitly, and `rq.addRequest` can be used too.

can we sync the number the same way as we do with `requestsFinished` and `requestsFailed`? i dont like this patching very much.